### PR TITLE
Add Sudo Mode for Admins

### PR DESCRIPTION
**Pull request by Sumukh**
An admin can su into another users account. This should be useful for looking at ok issues or viewing code (though it's intended for site admins - not staff because there are no course level restrictions). 

The way to access it is to login as a user and then go to `server-url.appspot.com/sudo/su/student@email.com`
It logs any attempts to sudo and displays a warning to admins. 

ok sudo 2015-01-25 23-11-13

I think this is a useful feature for admins and it should work with no further changes.

### CODE REVIEW

**Comment by Some3141** [comment]
Looks good to me- can you add some tests to make sure that this can't be abused in any way? Your changes look ok to me, but I just want to be sure that future modifications don't break this since it's a very easy target for security leaks.

**Comment by Sumukh** [comment]
@Some3141 - Pushed some tests - but I haven't figured out how to get the login working so the tests fail. Can you take a look? 

Right now it fails because it never logins - and the GET doesn't see a current_user.

**Comment by Some3141** [comment]
Is there a reason that you're not using the assertStatusCode method here: https://github.com/Cal-CS-61A-Staff/ok/blob/c665a9b2c08683b7dec29e23fc20fdb6c22c1aa3/server/tests/test_base.py#L200?

**Comment by Sumukh** [comment]
No great reason. I tried moving it into this class to see if it change the result - mainly because the responses being testing in this class should never be json data.

File: server/tests/integration/test_sudo.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                email=["dummy3@admin.com"],
+                is_admin=True
+            )
+        }
+
+    def get(self, url, *args, **kwds):
+        """
+        Makes a get request.
+        """
+        url = url
```
Comment (moowiz) [inline server/tests/integration/test_sudo.py]: Why is this here?

File: server/tests/integration/test_sudo.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    def testLoginToSudo(self):
+        self.login('admin')
+        self.get('/sudo/su/dummy@student.com')
+        self.assertStatusCode(200)
+        self.logout()
+
+    def testStudentAttemptToSudo(self):
+        self.login('student1')
+        self.get('/sudo/su/dummy@student.com')
+        self.assertStatusCode(404)
```
Comment (moowiz) [inline server/tests/integration/test_sudo.py]: We want a student trying to sudo to 404? Why, exactly?

File: server/tests/integration/test_sudo.py (hunk 1 above)
Comment (Sumukh) [inline server/tests/integration/test_sudo.py]: That line shouldn't be there. I brought in the get function because this test shouldn't be making json requests.

File: server/tests/integration/test_sudo.py (hunk 2 above)
Comment (Sumukh) [inline server/tests/integration/test_sudo.py]: It should be a 403. I think I used 404 because I hadn't implemented the 403 listener.

**Comment by moowiz** [comment]
Don't change the assertStatusCode. It needs to be different for `test_sudo.py`.

**Comment by Sumukh** [comment]
LGTM. Will merge and deploy with the rest of the admin features.
